kotsadm/pkg/handlers/gitops.go

```diff
@@ -105,6 +110,129 @@ func DisableAppGitOps(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	d, err := downstream.Get(clusterID)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	downstreamGitOps, err := gitops.GetDownstreamGitOps(a.ID, d.ClusterID)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if downstreamGitOps == nil {
+		logger.Error(errors.New("downstream gitops not found"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if downstreamGitOps.Format != "single" {
+		logger.Error(errors.New("unsupported gitops format"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err := gitops.TestGitOpsConnection(downstreamGitOps); err != nil {
+		logger.Error(err)
+		err = gitops.SetGitOpsError(a.ID, d.ClusterID, err.Error())
+		if err != nil {
+			logger.Error(err)
+		}
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err := gitops.SetGitOpsError(a.ID, d.ClusterID, ""); err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	go func(appID, clusterID, appSlug, appName, downstreamName string) {
+		currentVersion, err := downstream.GetCurrentVersion(appID, clusterID)
+		if err != nil {
+			logger.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		pendingVersions, err := downstream.GetPendingVersions(appID, clusterID)
+		if err != nil {
+			logger.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		// Create git commit for current version
+		currentVersionArchive, err := version.GetAppVersionArchive(appID, currentVersion.ParentSequence)
+		if err != nil {
+			logger.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		defer os.RemoveAll(currentVersionArchive)
+
+		_, err = gitops.CreateGitOpsCommit(downstreamGitOps, appSlug, appName, int(currentVersion.ParentSequence), currentVersionArchive, downstreamName)
+		if err != nil {
+			logger.Error(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		// Sort pending versions ascending before creating commits
+		sort.Slice(pendingVersions, func(i, j int) bool {
+			return pendingVersions[i].ParentSequence < pendingVersions[j].ParentSequence
+		})
+		// Create git commits for sorted pending versions
+		for _, pendingVersion := range pendingVersions {
+			pendingVersionArchive, err := version.GetAppVersionArchive(appID, pendingVersion.ParentSequence)
+			if err != nil {
+				logger.Error(err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			defer os.RemoveAll(pendingVersionArchive)
+
+			_, err = gitops.CreateGitOpsCommit(downstreamGitOps, appSlug, appName, int(pendingVersion.ParentSequence), pendingVersionArchive, downstreamName)
+			if err != nil {
+				logger.Error(err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}
+	}(a.ID, d.ClusterID, a.Slug, a.Name, d.Name)
```

Comment:
it's legal to use `a` inside the go routine.  you don't need to pass these in as arguments.  your call on this one.